vertical-pod-autoscaler/pkg/updater/eviction/pods_eviction_restriction.go

```
@@ -391,3 +437,199 @@ func setUpInformer(kubeClient kube_client.Interface, kind controllerKind) (cache
+		p, err := calculator.CalculatePatches(podToUpdate, vpa)
+		if err != nil {
+			klog.ErrorS(err, "failed to calculate resource patch for pod", "pod", klog.KObj(podToUpdate))
+			return err
+		}
+		klog.V(4).InfoS("Calculated patches for pod", "pod", klog.KObj(podToUpdate), "patches", p)
+		// TODO(maxcao13): change how this works later, this is gross and depends on the resource calculator being first in the slice
+		// we may not even want the updater to patch pod annotations at all
+		if i == 0 {
+			resourcePatches = append(resourcePatches, p...)
+		} else {
+			annotationPatches = append(annotationPatches, p...)
+		}
+	}
+	if len(resourcePatches) > 0 {
+		patch, err := json.Marshal(resourcePatches)
+		if err != nil {
+			klog.ErrorS(err, "Cannot marshal the patch %v", resourcePatches)
+			return err
+		}
+
+		res, err := e.client.CoreV1().Pods(podToUpdate.Namespace).Patch(context.TODO(), podToUpdate.Name, k8stypes.JSONPatchType, patch, metav1.PatchOptions{}, "resize")
+		if err != nil {
+			klog.ErrorS(err, "Failed to patch pod", "pod", klog.KObj(podToUpdate))
+			return err
+		}
+		klog.V(4).InfoS("In-place patched pod /resize subresource using patches ", "pod", klog.KObj(res), "patches", string(patch))
+
+		// TODO(maxcao13): whether or not we apply annotation patches should depend on resource patches?
+		if len(annotationPatches) > 0 {
+			patch, err := json.Marshal(annotationPatches)
+			if err != nil {
+				klog.ErrorS(err, "Cannot marshal the patch %v", annotationPatches)
+				return err
+			}
+			res, err = e.client.CoreV1().Pods(podToUpdate.Namespace).Patch(context.TODO(), podToUpdate.Name, k8stypes.JSONPatchType, patch, metav1.PatchOptions{})
+			if err != nil {
+				klog.ErrorS(err, "Failed to patch pod", "pod", klog.KObj(podToUpdate))
+				return err
+			}
+			klog.V(4).InfoS("Patched pod annotations", "pod", klog.KObj(res), "patches", string(patch))
+		}
+	} else {
+		err := fmt.Errorf("no patches to apply to %s", podToUpdate.Name)
```

Can we combine it to one? Maybe something like that:
```go
klog.ErrorS("Failed to patch pod. no patches to apply", "pod", klog.KObj(podToUpdate))
```

---

I'm not sure if that would work, I still need to return an err. Maybe this is better? WDYT?

```go
err := fmt.Errorf("no resource patches were calculated to apply")
klog.ErrorS(err, "Failed to patch pod", "pod", klog.KObj(podToUpdate))
return err
```

---

Why do we need to log the error inside the function if the function is already returning an error that gets logged here:
https://github.com/kubernetes/autoscaler/blob/53d4006782a8f1ec8513cef37a49ba8f3662019d/vertical-pod-autoscaler/pkg/updater/logic/updater.go#L469
maybe those logs are redundant?

---

Yeah, you're probably right. I'll just remove the log here, thanks!